spec.bs

```diff
@@ -632,9 +609,207 @@ To <dfn>obtain a report's shared info</dfn> given an [=aggregatable report=]
+1. [=Upon fulfillment=] of |p| or [=upon rejection=] of |p|, run the following
+    steps:
+    1. TODO: Take onEvent contributions from its cache, run [=fill in the
+        contribution=] and then [=append an entry to the contribution cache=].
+    1. [=list/iterate|For each=] |origin| → |batchingScope| of
+        |batchingScopeMap|:
+        1. [=Process contributions for a batching scope=] given |batchingScope|,
+            |origin| and "<code>protected-audience</code>".
+
+</div>
+
+Issue: How to handle fenced frame-triggered contributions and other
+    event-triggered contributions.
+
+Issue: Need to handle `auctionReportBuyers` and `auctionReportBuyerKeys` here or
+    in the Protected Audience API spec.
+
+The <a spec="turtledove">evaluate a script</a> steps are modified to add the
+following steps after step 11 ("If <var ignore>evaluationStatus</var> is an
+[=ECMAScript/abrupt completion=]..."), renumbering later steps as appropriate:
+<div algorithm="protected-audience-evaluate-a-script-monkey-patch">
+12. Set <var ignore>global</var>'s {{InterestGroupScriptRunnerGlobalScope/
+    privateAggregation}}'s [=PrivateAggregation/get batching scope steps=] to
+    the following algorithm:
+    1. Return the [=batching scope=] associated with the auction and the
+        <var ignore>scope</var>'s [=relevant settings object=]'s [=environment
+        settings object/origin=].
+
+</div>
+
+Issue: Once <a href="https://github.com/wicg/turtledove/issues/615">
+    protected-audience/615</a> is resolved, align the above monkey patch with
+    how access to other functions is prevented in
+    {{InterestGroupScriptRunnerGlobalScope}}s until the script's initial
+    execution is complete.
+
+<div algorithm>
+The <dfn method for="PrivateAggregation">contributeToHistogramOnEvent(DOMString
+event, PAExtendedHistogramContribution contribution)</dfn> method steps are:
+</div>
+
+Issue: Fill in validation. Will need to check offsets are valid and in the right
+    range for the type etc. Also that base values are one of a set.
+
+Issue: Need to document limits on offset, etc.
+
+Issue: Fill in the rest. (Need to put the contribution in some sort of queue and
+    process the queue at some point. Need to decide where the queue should live
+    given it has to outlive the auction.)
+
+Issue(44): Consider accepting an array of contributions.
+
+Protected Audience API-specific structures {#protected-audience-api-specific-structures}
+----------------------------------------------------------------------------------------
+
+<h4 dfn-type=dfn>Signal base value</h3>
+A signal base value is one of the following:
+<dl dfn-for="signal base value">
+: "<dfn export><code>winning-bid</code></dfn>"
+:: The bid value of the winning bid.
+: "<dfn export><code>highest-scoring-other-bid</code></dfn>"
+:: The bid value of the highest scoring bid that did not win.
+: "<dfn export><code>script-run-time</code></dfn>"
+:: The running time of the script in ms(?).
```

Comment:
Could we add an Issue to confirm this?

Reply:
Just confirmed -- removed the uncertainty here